lib/builder/builder.ex

```diff
@@ -0,0 +1,160 @@
+
+    # look for override target in system env
+    # if it's a valid target, set it as the only target
+    target_override_string = System.get_env("BURRITO_TARGET")
+
+    build_targets =
+      if target_override_string do
+        Log.warning(
+          :build,
+          "Target is being overridden with BURRITO_TARGET #{target_override_string}"
+        )
+
+        old_targets = Target.get_old_targets()
+
+        override_atom = try do
+          String.to_existing_atom(target_override_string)
+        rescue
+          _ -> exit_invalid_target(target_override_string)
+        end
+
+        # If we have a named target defined that matches this atom use that
+        # otherwise if it's a legacy build target, translate it (this will be removed soon!)
+        # otherwise :error, not a valid target
+        cond do
+          Keyword.has_key?(build_targets, override_atom) ->
+            Keyword.take(build_targets, [override_atom])
+
+          override_atom in old_targets ->
+            resolved_override = Target.maybe_translate_old_target(override_atom)
+            Keyword.put([], override_atom, resolved_override)
+
+          true ->
+            exit_invalid_target(override_atom)
+        end
+      else
+        build_targets
+      end
+
+    # Build every target
+    Enum.each(build_targets, fn {name, t} ->
+      target = Target.init_target(t, name, debug?)
+
+      self_path =
+        __ENV__.file
+        |> Path.dirname()
+        |> Path.split()
+        |> List.delete_at(-1)
+        |> List.delete_at(-1)
+        |> Path.join()
+
+      initial_context = %Context{
+        target: target,
+        erts_location: :local,
+        cross_build: false,
+        mix_release: release,
+        work_dir: "",
+        self_dir: self_path,
+        halt: false
+      }
+
+      Log.info(:build, "Burrito is building target: #{target.alias}")
+
+      Log.info(
+        :build,
+        "Burrito will build for target:\n\tOS: #{target.os}\n\tCPU: #{target.cpu}\n\tQualifiers: #{inspect(target.qualifiers)}\n\tDebug: #{target.debug?}"
+      )
+
+      Enum.reduce(@phases, initial_context, &run_phase/2)
+    end)
+
+    # All done!
+    release
+  end
+
+  defp run_phase({phase_name, mod_list}, %Context{} = context) do
+    Log.info(:phase, "PHASE: #{inspect(phase_name)}")
+
+    # Load in extra steps, pre and post
+    extra_steps = context.mix_release.options[:burrito][:extra_steps]
+    extra_steps_pre = extra_steps[phase_name][:pre] || []
+    extra_steps_post = extra_steps[phase_name][:post] || []
+
+    mod_list = extra_steps_pre ++ mod_list ++ extra_steps_post
+
+    Enum.reduce(mod_list, context, fn mod, %Context{} = acc ->
+      %Context{} = new_context = mod.execute(acc)
+
+      # Halt if `halt` flag was set
+      if new_context.halt do
+        Log.error(
+          :build,
+          "Halt requested from phase: #{inspect(phase_name)} in step #{inspect(mod)}"
+        )
+
+        exit(1)
```

Comment:
Is there any risk of immediately halting here, without doing any sort of cleanup first?

Reply:
This comment is further explored here: https://github.com/burrito-elixir/burrito/pull/30/files#r770180010